### DISCUSSION TITLE
Is Zustand 3.7.1 Compatible with React 18?

### DISCUSSION THREAD

**Discussion by rortan134** [pmndrs/zustand#885] [Q&A] [answered]
I'm thinking about upgrading my app to React 18 to take advantage of the new features, though its state is completely built on top of zustand, is it compatible?

**Comment by dai-shi** [comment, marked as answer]
In practice, v3 should work almost fine with React 18. v4 will use the new hook `useSyncExternalStore` in React 18 to avoid tearing in some edge cases.
Out of curiosity, what are the new features you would like to take advantage of?

**Reply by rortan134** [reply]
Mostly the useTransition, startTransition together with concurrent rendering features to create a better user experience since my app requires large processing rerendering time (an algorithm) and I'd like other things to not be completely paused while it's running.

**Reply by dai-shi** [reply]
Well, startTransition is probably the trickiest one.
useDefferedValue would work better with useSyncExternalStore.
See this for more technical details: https://github.com/reactwg/react-18/discussions/86

**Reply by marcellomontemagno** [reply]
I've been quite happy with zustand so far but this might be a problem in my current app as I have plenty of use cases for long running client-side filters where the `isPending` flag from `useTransition` would be handy. As a side note, `useDeferredValue` seems not as helpful when rendering a lot of items in a single pass (this is quite common when results change dramatically during filtering), I'm currently trying to figure out if I'm missing something about this.

**Reply by dai-shi** [reply]
use-zustand https://twitter.com/dai_shi/status/1581466494973595649
